plugins/follow_me/follow_me.h

```diff
@@ -0,0 +1,206 @@
+#pragma once
+
+#include <iostream>
+#include <cmath>
+
+
+namespace dronecore {
+
+class FollowMeImpl;
+
+class FollowMe
+{
+public:
+    explicit FollowMe(FollowMeImpl *impl);
+    ~FollowMe();
+
+    /**
+     * @brief Results of FollowMe operations
+     */
+    enum class Result {
+        SUCCESS = 0, /**< @brief %Request succeeded. */
+        NO_DEVICE, /**< @brief No device connected. */
+        CONNECTION_ERROR, /**< @brief %Connection error. */
+        BUSY, /**< @brief Vehicle busy. */
+        COMMAND_DENIED, /**< @brief Command denied. */
+        TIMEOUT, /**< @brief %Request timeout. */
+        UNKNOWN /**< @brief Unknown error. */
+    };
+
+    /**
+     * @brief FollowMe configuration.
+     * Reference: https://docs.px4.io/en/advanced_config/parameter_reference.html#follow-target
+     */
+    class Configuration
+    {
+    public:
+        /**
+         * @brief Side to follow target from
```

Comment:
Looks good, but all sentences should terminate in a full stop, everywhere.

Reply:
@hamishwillee  Thanks. I've a small doubt about best practice of configuration class. Generally Configuration or Options classes are not nested in another class. For example `class FollowMeConfiguration`, can be external to `class FollowMe`.  I am not sure what is the right way here.

Reply:
Hi @shakthi-prashanth-m 

Having classes in classes makes perfect sense where the definition is intimately associated with the parent and doesn't need to be shared around outside of the parent. That is IMO the case here. 

But @julianoes owns the broad API definition, so lets see what he says.